test/format/import.c

```diff
@@ -0,0 +1,212 @@
+static void import_with_repair(WT_SESSION *session);
+static void import_with_file_metadata(WT_SESSION *session, WT_SESSION *import_session);
+static void verify_import(WT_SESSION *session, int start_value);
+
+/*
+ * Import directory initialize command, remove and re-create the primary backup directory, plus a
+ * copy we maintain for recovery testing.
+ */
+#define HOME_IMPORT_INIT_CMD "rm -rf %s/" IMPORT_DIR "&& mkdir %s/" IMPORT_DIR
+#define IMPORT_DIR "IMPORT"
+#define IMPORT_URI "table:import"
+#define IMPORT_URI_CONFIG "key_format=i,value_format=i"
+#define IMPORT_ENTRIES 1000
+
+/*
+ * import --
+ *     Periodically import table.
+ */
+WT_THREAD_RET
+import(void *arg)
+{
+    WT_CONNECTION *conn, *import_conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION *import_session, *session;
+    size_t cmd_len;
+    uint32_t import_value;
+    u_int period;
+    int counter;
+    char cmd[64];
+
+    WT_UNUSED(arg);
+    conn = g.wts_conn;
+    counter = 0;
+    import_value = false;
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) * 2 + strlen(HOME_IMPORT_INIT_CMD) + 1;
+    testutil_check(__wt_snprintf(cmd, cmd_len, HOME_IMPORT_INIT_CMD, g.home, g.home));
+    testutil_checkfmt(system(cmd), "%s", "import directory creation failed");
+
+    memset(cmd, 0, sizeof(cmd));
+    cmd_len = strlen(g.home) + strlen(IMPORT_DIR) + 10;
+    testutil_check(__wt_snprintf(cmd, cmd_len, "%s/%s", g.home, IMPORT_DIR));
+    /* Open a connection to the database, creating it if necessary. */
+    create_database(cmd, &import_conn);
+
+    /* Open a session */
+    testutil_check(import_conn->open_session(import_conn, NULL, NULL, &import_session));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    testutil_checkfmt(
+      import_session->create(import_session, IMPORT_URI, IMPORT_URI_CONFIG), "%s", g.uri);
+
+    testutil_check(import_session->open_cursor(import_session, IMPORT_URI, NULL, NULL, &cursor));
+    while (!g.workers_finished) {
+        period = mmrand(NULL, 1, 10);
+
+        for (int i = 0; i < IMPORT_ENTRIES; ++i) {
+            cursor->set_key(cursor, i);
+            cursor->set_value(cursor, counter + i);
+            testutil_check(cursor->insert(cursor));
+        }
+        counter += IMPORT_ENTRIES;
+        testutil_check(import_session->checkpoint(import_session, NULL));
+
+        /* Copy table into current test/format directory */
+        copy_file_into_directory(session, IMPORT_DIR, "import.wt");
+
+        import_value = mmrand(NULL, 0, 1);
+        if (import_value == 0) {
+            import_with_repair(session);
+        } else {
+            import_with_file_metadata(session, import_session);
+        }
+
+        /* Drop import table, so we can import next iteration */
+        while ((ret = session->drop(session, IMPORT_URI, NULL)) == EBUSY) {
+            __wt_yield();
+        }
+        testutil_check(ret);
+
+        verify_import(import_session, counter - IMPORT_ENTRIES);
```

Comment:
I think this is not correct. Before we drop the table in the test database, we want to verify the contents of the newly imported table in the test database. That makes sure we can read it and get to all the expected data. That is the table that is of interest. So you want to:
1. Move this call before `drop`.
2. Send in `session` instead.
3. (If you take my suggestion you should not need the 2nd arg.)

Reply:
oops, thanks for catching it! I was meant to verify on the the normal session